src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Connection.cs

```diff
@@ -272,14 +271,32 @@ private void OnRead(UvStreamHandle handle, int status)
                 }
 
                 error = new IOException(uvError.Message, uvError);
+                _currentWritableBuffer?.Commit();
             }
+            else
+            {
+                Debug.Assert(_currentWritableBuffer != null);
 
-            Input.IncomingComplete(readCount, error);
+                var currentWritableBuffer = _currentWritableBuffer.Value;
+                currentWritableBuffer.Advance(readCount);
+                var flushTask = currentWritableBuffer.FlushAsync();
+                if (!flushTask.IsCompleted)
+                {
+                    ((IConnectionControl) this).Pause();
+                    flushTask.ContinueWith((task, state) =>
+                    {
+                        Thread.Post((connectionControl) => ((IConnectionControl) connectionControl).Resume(), state);
```

Comment:
Shouldn't we not resume if task.Result is false indicating that the consumer is gone? I get in this case we should be closing the socket anyway, but still.